crates/sui-indexer-alt-graphql/src/api/types/object.rs

```diff
@@ -422,25 +420,33 @@ impl Object {
         version: Option<UInt53>,
         root_version: Option<UInt53>,
         checkpoint: Option<UInt53>,
-    ) -> Result<Option<Self>, RpcError<Error>> {
+    ) -> Option<Result<Self, RpcError<Error>>> {
         let key = ObjectKey {
             address: self.super_.address.into(),
             version,
             root_version,
             at_checkpoint: checkpoint,
         };
 
-        Object::by_key(ctx, self.super_.scope.without_root_version(), key).await
+        Object::by_key(ctx, self.super_.scope.without_root_version(), key)
+            .await
+            .transpose()
     }
 
     /// The Base64-encoded BCS serialization of this object, as an `Object`.
-    pub(crate) async fn object_bcs(&self, ctx: &Context<'_>) -> Result<Option<Base64>, RpcError> {
-        let Some(object) = self.contents(ctx).await?.as_ref() else {
-            return Ok(None);
+    pub(crate) async fn object_bcs(&self, ctx: &Context<'_>) -> Option<Result<Base64, RpcError>> {
+        let object = match self.contents(ctx).await {
+            Ok(Some(o)) => o,
+            Ok(None) => return None,
+            Err(e) => return Some(Err(e)),
         };
```

Comment:
This is actually similar to the pattern I described above (and I do prefer it to using `transpose`, but you can avoid having to pattern match on the outer `Option<...>` by using `?` syntax.


```suggestion
        let object = match self.contents(ctx).await.transpose()? {
            Ok(o) => o,
            Err(e) => return Some(Err(e)),
};
```